jax/random.py

```diff
@@ -1116,15 +1116,15 @@ def _poisson(key, lam, shape, dtype):
   # The acceptance probability for rejection sampling maxes out at 89% as
   # λ -> ∞, so pick some arbitrary large value.
   lam_rejection = lax.select(use_knuth, lax.full_like(lam, 1e5), lam)
-  max_iters = jnp.iinfo(dtype).max  # insanely conservative
+  max_iters = dtype.type(jnp.iinfo(dtype).max)  # insanely conservative
```

Comment:
This prevents type mismatches in lax under certain circumstances revealed by my x64 explorations.

Reply:
(See a description of the issue this fixes in https://github.com/google/jax/issues/3840#issuecomment-663630793)